docs/architecture/grpc-for-wcf-developers/migrating-request-reply.md

```diff
@@ -0,0 +1,418 @@
+                Holding = source.Holding,
+                Cost = Convert.ToDouble(source.Cost)
+            };
+        }
+    }
+
+    public partial class Portfolio
+    {
+        public static Portfolio FromRepositoryModel(PortfolioData.Models.Portfolio source)
+        {
+            if (source is null) return null;
+
+            var target = new Portfolio
+            {
+                Id = source.Id,
+                TraderId = source.TraderId.ToString(),
+            };
+
+            target.Items.AddRange(source.Items.Select(PortfolioItem.FromRepositoryModel));
+
+            return target;
+        }
+    }
+}
+```
+
+> [!NOTE]
+> You could use a library like [AutoMapper](https://automapper.org/) to handle this conversion from internal model classes to Protobuf types, as long as you configure the lower-level type conversions like `string`/`Guid` or `decimal`/`double` and the list mapping.
+
+With the conversion code in place, the `Get` method implementation can be completed.
+
+```csharp
+public override async Task<GetResponse> Get(GetRequest request, ServerCallContext context)
+{
+    if (!Guid.TryParse(request.TraderId, out var traderId))
+    {
+        throw new RpcException(new Status(StatusCode.InvalidArgument, "traderId must be a UUID"));
+    }
+
+    var portfolio = await _repository.GetAsync(traderId, request.PortfolioId);
+
+    return new GetResponse
+    {
+        Portfolio = Portfolio.FromRepositoryModel(portfolio)
+    };
+}
+
+```
+
+The implementation of the `GetAll` method is similar. Note that the `repeated` fields on Protobuf messages are generated as `readonly` properties of type `RepeatedField<T>`, so you have to add items to them using the `AddRange` method.
+
+```csharp
+public override async Task<GetAllResponse> GetAll(GetAllRequest request, ServerCallContext context)
+{
+    if (!Guid.TryParse(request.TraderId, out var traderId))
+    {
+        throw new RpcException(new Status(StatusCode.InvalidArgument, "traderId must be a UUID"));
+    }
+
+    var portfolios = await _repository.GetAllAsync(traderId);
+
+    var response = new GetAllResponse();
+    response.Portfolios.AddRange(portfolios.Select(Portfolio.FromRepositoryModel));
+
+    return response;
+}
+```
+
+Having successfully migrated the WCF application to gRPC, let's look at creating a client for it from the `.proto` file.
+
+## Generating Client code
+
+Create a .NET Standard class library in the same solution to contain the client. This is primarily as an example of creating client code, but such a library could be packaged using NuGet and distributed on an internal repository for other .NET teams to consume. Go ahead and add a new .NET Standard Class Library called `TraderSys.Portfolios.Client` to the solution and delete the `Class1.cs` file.
+
+> [!CAUTION]
+> The latest `Grpc.Net` NuGet packages (0.2.23 or later) require .NET Standard 2.1. You might need to change your Target Framework to `netstandard2.1` for the new class library.
```

Comment:
I don't think the change is worth the warning. Previously the client packages needed netcoreapp3.0 which was less compatible.

I think a generic note that Grpc.Net.Client requires .NET Core 3 or a .NET Standard 2.1 compatible runtime. (which I don't think there are any other than .NET Core 3)